src/store.rs

```diff
@@ -1169,633 +908,521 @@ impl<F: LurkField> Store<F> {
         }
     }
 
-    /// Mutable version of car_cdr to handle Str. `(cdr str)` may return a new str (the tail), which must be allocated.
-    pub fn car_cdr_mut(&mut self, ptr: &Ptr<F>) -> Result<(Ptr<F>, Ptr<F>), Error> {
-        match ptr.tag {
-            ExprTag::Nil => Ok((self.get_nil(), self.get_nil())),
-            ExprTag::Cons => match self.fetch(ptr) {
-                Some(Expression::Cons(car, cdr)) => Ok((car, cdr)),
-                Some(Expression::Opaque(_)) => Err(Error("cannot destructure opaque Cons".into())),
-                _ => unreachable!(),
-            },
-            ExprTag::Str => {
-                if let Some(Expression::Str(s)) = self.fetch(ptr) {
-                    let mut str = s.chars();
-                    if let Some(c) = str.next() {
-                        let cdr_str: String = str.collect();
-                        let str = self.intern_str(&cdr_str);
-                        Ok((self.get_char(c), str))
-                    } else {
-                        Ok((self.nil(), self.intern_str("")))
-                    }
-                } else {
-                    panic!();
+    pub fn get_z_cont(
+        &self,
+        ptr: &ContPtr<F>,
+        z_store: Option<Rc<RefCell<ZStore<F>>>>,
+    ) -> Result<(ZContPtr<F>, Option<ZCont<F>>), Error> {
+        if let Some(idx) = ptr.raw.opaque_idx() {
+            let z_ptr = self
+                .opaque_cont_ptrs
+                .get_index(idx)
+                .ok_or(Error("get_z_expr unknown opaque".into()))?;
+            Ok((*z_ptr, None))
+            // TODO: should we try to dereference the opaque ptr here?
```

Comment:
Probably. If the question is, "What happens when an apparently opaque pointer actually has a known value?" then I think the answer should be, "we find the real value." We don't have a well-defined concept of 'explicitly opaque' pointers. That is what commitments are for.

Reply:
Punting this for a future PR per our discussion in the meeting